src/components/ha-sidebar.ts

```diff
@@ -179,23 +250,72 @@ class HaSidebar extends LitElement {
         @focusout=${this._listboxFocusOut}
         @scroll=${this._listboxScroll}
         @keydown=${this._listboxKeydown}
+        @action=${this._handleAction}
+        .actionHandler=${actionHandler({
+          hasHold: !this._editMode,
+        })}
       >
-        ${this._renderPanel(
-          defaultPanel.url_path,
-          defaultPanel.title || hass.localize("panel.states"),
-          defaultPanel.icon,
-          !defaultPanel.icon ? mdiViewDashboard : undefined
-        )}
-        ${beforeSpacer.map((panel) =>
-          this._renderPanel(
-            panel.url_path,
-            hass.localize(`panel.${panel.title}`) || panel.title,
-            panel.icon,
-            undefined
-          )
-        )}
+        ${this._editMode
+          ? html`<div id="sortable">
+              ${guard([this._hiddenPanels, this._renderEmptySortable], () =>
+                this._renderEmptySortable
+                  ? ""
+                  : beforeSpacer.map((panel) =>
+                      this._renderPanel(
+                        panel.url_path,
+                        panel.url_path === "lovelace"
+                          ? hass.localize("panel.states")
+                          : hass.localize(`panel.${panel.title}`) ||
+                              panel.title,
+                        panel.url_path === "lovelace" ? undefined : panel.icon,
+                        panel.url_path === "lovelace"
+                          ? mdiViewDashboard
+                          : undefined
+                      )
+                    )
+              )}
+            </div>`
+          : html`${beforeSpacer.map((panel) =>
+              this._renderPanel(
+                panel.url_path,
+                panel.url_path === "lovelace"
+                  ? hass.localize("panel.states")
+                  : hass.localize(`panel.${panel.title}`) || panel.title,
+                panel.url_path === "lovelace" ? undefined : panel.icon,
+                panel.url_path === "lovelace" ? mdiViewDashboard : undefined
+              )
```

Comment:
Since this is the same logic as above, can we extract it into a function to be reusable?